src/platform-apps/channels/channel-view.tsx

```diff
@@ -164,7 +152,15 @@ export class ChannelView extends React.Component<Properties, State> {
           </div>
           {this.props.messages.length > 0 && <Waypoint onEnter={this.props.onFetchMore} />}
           {this.props.messages.length > 0 && this.renderMessages()}
-          {this.renderChatWindow()}
+          <IfAuthenticated show>
+            <MessageInput
+              placeholder='Speak your truth...'
+              onSubmit={this.props.sendMessage}
+            />
+          </IfAuthenticated>
+          <IfAuthenticated hide>
+            <ConnectButton />
+          </IfAuthenticated>
```

Comment:
instead of needing to have multiple blocks, and add the ConnectButton everywhere you want to show it, i think that part can just be pulled into the IfAuthenticated component. something like we do in the blockchain restricted component:
https://github.com/m3m3n70/zero-web/blob/development/src/components/blockchain-restricted/container.tsx

i think it will be a pretty common pattern to either show content, or show the connect button. so adding a `includeConnectButton` property or something similar to the IfAuthenticated component would streamline that.